### ISSUE TITLE
Display legend in plot

### ISSUE THREAD

**Issue by UUSim** [silx-kit/silx#1447]
How can I display a legend with all curves in a PlotWidget?

(Similar to the legend functionality in matplotlib? [Legend guide](https://matplotlib.org/users/legend_guide.html))

**Comment by vallsv** [comment]
Hi,

We don't have this concept on legend in silx, but the default Plot1d provide a widget containing the list of the curves. You can find it on the bottom of the widget: `Options`, then check `Legends`.
![screenshot plotwidget legend](https://user-images.githubusercontent.com/7579321/34674630-9540959c-f47e-11e7-9bd1-5d037b452091.png)

If you use a real `PlotWidget` you can create your legend in this way:
```
import silx.gui.plot
plot = silx.gui.plot.PlotWidget()
plot.addCurve([1,2,3],[0,1,2],legend="foo")
legend = silx.gui.plot.LegendSelector.LegendsDockWidget(plot=plot)

plot.show()
legend.show()
```

If it is not enough for you we can show you how to hack the matplotlib backend of our widgets. But this part will not be stable.

**Comment by UUSim** [comment]
Thanks a lot for the explanation. This is not exactly what I'm looking for, although it might be useable after some tweaking:
- Reducing the visual space taken by the widget.
- Allow for curve aliases (show a different name for a curve)
- Ideally, the legend should be included when performing an export to png.

On a side note: We are using the `opengl` backend, so using the matplotlib legend is not an option.

**Comment by vasole** [comment]
If I am not mistaken, the legends widget support a context menu allowing to rename a curve.

**Comment by vasole** [comment]
In PyMca one distinguish display and final output. For the later one allows full customization. I cannot imagine the user customizing everything for each plot.

From this plot:

![image](https://user-images.githubusercontent.com/5636251/42622501-d2857b10-85c0-11e8-8149-ac9f9383045e.png)

One reaches this one where everything can be customized for the final publication quality output.

![image](https://user-images.githubusercontent.com/5636251/42622643-24e6d9f8-85c1-11e8-8a22-e9cd9938f08b.png)

**Comment by UUSim** [comment]
This issue is fixed since the PR #2041 has been merged and released.

Thanks guys!
